spec/lib/rmt/cli/products_spec.rb

```diff
@@ -468,4 +468,130 @@
+      output = "Product: #{product.friendly_name} (id: #{target})\n"
+      output += "Description: #{product.description}\n"
+      output += "Repositories:\n"
+      repos.each do |repo|
+        output += "* #{repo.name} (id: #{repo.scc_id})"\
+          " (non-mandatory, not enabled, not mirrored)\n"
+      end
+      output
+    end
+
+    describe 'success' do
+      context 'with disabled and not mirrored repositories' do
+        it 'shows details of a product and repositories' do
+          expect { described_class.start(argv) }
+            .to output(expected_output).to_stdout
+            .and output('').to_stderr
+        end
+      end
+
+      context 'with disabled and mirrored repositories' do
+        let(:product) { create :product, :with_disabled_mirrored_repositories }
+        let(:expected_output) do
+          output = "Product: #{product.friendly_name} (id: #{target})\n"
+          output += "Description: #{product.description}\n"
+          output += "Repositories:\n"
+          repos.each do |repo|
+            output += "* #{repo.name} (id: #{repo.scc_id})"\
+              " (non-mandatory, enabled, not mirrored)\n"
+          end
+          output
+        end
+
+        it 'shows details of a product and repositories' do
+          expect { described_class.start(argv) }
+            .to output(expected_output).to_stdout
+            .and output('').to_stderr
+        end
+      end
+
+      context 'with enabled and not mirrored repositories' do
+        let(:product) { create :product, :with_not_mirrored_repositories }
+        let(:expected_output) do
+          output = "Product: #{product.friendly_name} (id: #{target})\n"
+          output += "Description: #{product.description}\n"
+          output += "Repositories:\n"
+          repos.each do |repo|
+            output += "* #{repo.name} (id: #{repo.scc_id})"\
+              " (mandatory, not enabled, not mirrored)\n"
+          end
+          output
+        end
+
+        it 'shows the details of a product and its repositories' do
+          expect { described_class.start(argv) }
+            .to output(expected_output).to_stdout
+            .and output('').to_stderr
+        end
+      end
+
+      context 'with enabled and mirrored repositories' do
+        let(:product) { create :product, :with_mirrored_repositories }
+        let(:expected_output) do
+          output = "Product: #{product.friendly_name} (id: #{target})\n"
+          output += "Description: #{product.description}\n"
+          output += "Repositories:\n"
+          repos.each do |repo|
+            output += "* #{repo.name} (id: #{repo.scc_id})"\
+              " (mandatory, enabled, not mirrored)\n"
+          end
+          output
+        end
+
+        it 'shows details of a product and its repositories' do
+          expect { described_class.start(argv) }
+            .to output(expected_output).to_stdout
+            .and output('').to_stderr
+        end
+      end
```

Comment:
It feels like there is a lot of repetition in the last 3 tests. Is there a way to do this more concise?

Reply:
@digitaltom Sure, I'm on it.